src/setfit/data.py

```diff
@@ -99,8 +99,8 @@ def sample_dataset(dataset: Dataset, label_column: str = "label", num_samples: i
     samples = []
     for label in range(num_labels):
         data = shuffled_dataset.filter(lambda example: int(example[label_column]) == label)
-        num_samples = min(len(data) // 2, num_samples)
-        samples.append(data.select([i for i in range(num_samples)]))
+        num_label_samples = min(len(data) // 2, num_samples)
```

Comment:
Good catch about the uneccesary integer division - `len(data)` should always be an `int` so this is probably what we want:
```suggestion
        num_label_samples = min(len(data) / 2, num_samples)
```

Reply:
Although `len(data)` is always an int, `len(data) / 2` is not, e.g. if `data` has a length of 3. 
That said, isn't the best solution simply `num_label_samples = min(len(data), num_samples)`?

Reply:
Yes, I think you're right! Let's switch to that for now :)

Reply:
Done!